### PR TITLE
chore(*) release Kuma 1.4.1

### PR DESCRIPTION
**Pull request by bartsmykla**
### Summary

Release the Kuma 1.4.1

### Full changelog

No changelog

### Issues resolved

No issues resolved

### Documentation

- [ ] Link to the website [documentation PR](https://github.com/kumahq/kuma-website/pull/XXX)

### Testing

- [ ] Unit tests
- [ ] E2E tests
- [ ] Manual testing on Universal
- [ ] Manual testing on Kubernetes

### Backwards compatibility

- [ ] Update [`UPGRADE.md`](UPGRADE.md) with any steps users will need to take
      when upgrading.
- [ ] Add `backport-to-stable` label if the code is backwards compatible. Otherwise, list breaking changes.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,49 @@
...
+* feat: add affinity to CP and Ingress pods [#3036](https://github.com//kumahq/kuma/pull/3036)
+  👍contributed by @andrey-dubnik
+* chore: bump github.com/gruntwork-io/terratest [#3433](https://github.com//kumahq/kuma/pull/3433)
+* chore: bump github.com/golang-jwt/jwt/v4 from 4.1.0 to 4.2.0 [#3432](https://github.com//kumahq/kuma/pull/3432)
+* feat: consolidate tokens logic to support expiration, rotation, revocation and RSA256
+* ci: fix failing mac CI because of unsupported instance size [#3408](https://github.com//kumahq/kuma/pull/3408)
+* fix:: simplify cluster creation with endpoints [#3403](https://github.com//kumahq/kuma/pull/3403)
+* fix: enable metrics hijacker for current version of Kuma [#3405](https://github.com//kumahq/kuma/pull/3405)
+* fix: switch to mTLS when CP communicates with Envoy Admin [#3353](https://github.com//kumahq/kuma/pull/3353)
+* test: disable KIC E2E tests on Kind [#3398](https://github.com//kumahq/kuma/pull/3398)
```
Comment (jakubdyszkiewicz) [inline CHANGELOG.md:22]: same

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,5 +1,49 @@
...
+* chore: bump github.com/spiffe/spire from 0.12.3 to 1.1.1 [#3388](https://github.com//kumahq/kuma/pull/3388)
+* chore: bump github.com/spf13/viper from 1.8.1 to 1.9.0 [#3389](https://github.com//kumahq/kuma/pull/3389)
+* fix: validate cp url in dp conf [#3357](https://github.com//kumahq/kuma/pull/3357)
+* chore: send reports to tls endpoint [#3361](https://github.com//kumahq/kuma/pull/3361)
+* chore: check explicit service account name [#3228](https://github.com//kumahq/kuma/pull/3228)
+* feat: inspect other dependencies versions [#3352](https://github.com//kumahq/kuma/pull/3352)
+* chore: upgrade minikube to v1.24.0 [#3269](https://github.com//kumahq/kuma/pull/3269)
```
Comment (jakubdyszkiewicz) [inline CHANGELOG.md:29]: same

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,5 +1,49 @@
...
+* chore: add area/gateway label [#3263](https://github.com//kumahq/kuma/pull/3263)
+* build: notify about merged PRs on release branch [#3289](https://github.com//kumahq/kuma/pull/3289)
```
Comment (jakubdyszkiewicz) [inline CHANGELOG.md:31]: same

File: CHANGELOG.md [hunk 4]
```diff
@@ -1,5 +1,49 @@
...
+* chore: remove dp token from xds metadata [#3282](https://github.com//kumahq/kuma/pull/3282)
+* refactor: move from io/ioutil to io and os packages [#3265](https://github.com//kumahq/kuma/pull/3265)
+  👍contributed by @Juneezee
+* build: github notify about merged PRs workflow [#3258](https://github.com//kumahq/kuma/pull/3258)
```
Comment (jakubdyszkiewicz) [inline CHANGELOG.md:35]: same

File: CHANGELOG.md [hunk 5]
```diff
@@ -1,5 +1,49 @@
...
+* fix: validate newly generated xDS snapshots [#3195](https://github.com//kumahq/kuma/pull/3195)
+* chore: remove RBAC generator [#3176](https://github.com//kumahq/kuma/pull/3176)
+* fix: run test/release also for RCs [#3208](https://github.com//kumahq/kuma/pull/3208)
```
Comment (jakubdyszkiewicz) [inline CHANGELOG.md:38]: same

File: CHANGELOG.md [hunk 6]
```diff
@@ -1,5 +1,49 @@
...
+* feat: inspect other dependencies versions [#3352](https://github.com//kumahq/kuma/pull/3352)
+* chore: upgrade minikube to v1.24.0 [#3269](https://github.com//kumahq/kuma/pull/3269)
+* chore: add area/gateway label [#3263](https://github.com//kumahq/kuma/pull/3263)
+* build: notify about merged PRs on release branch [#3289](https://github.com//kumahq/kuma/pull/3289)
+* chore: remove dp token from xds metadata [#3282](https://github.com//kumahq/kuma/pull/3282)
+* refactor: move from io/ioutil to io and os packages [#3265](https://github.com//kumahq/kuma/pull/3265)
+  👍contributed by @Juneezee
+* build: github notify about merged PRs workflow [#3258](https://github.com//kumahq/kuma/pull/3258)
+* fix: validate newly generated xDS snapshots [#3195](https://github.com//kumahq/kuma/pull/3195)
+* chore: remove RBAC generator [#3176](https://github.com//kumahq/kuma/pull/3176)
```
Comment (jakubdyszkiewicz) [inline CHANGELOG.md:37]: same - it's code cleanup

File: CHANGELOG.md [hunk 7]
```diff
@@ -1,5 +1,49 @@
...
+* fix: run test/release also for RCs [#3208](https://github.com//kumahq/kuma/pull/3208)
+* chore: bump k8s.io/apiextensions-apiserver from 0.22.3 to 0.22.4 [#3218](https://github.com//kumahq/kuma/pull/3218)
+* chore: bump github.com/testcontainers/testcontainers-go [#3221](https://github.com//kumahq/kuma/pull/3221)
```
Comment (jakubdyszkiewicz) [inline CHANGELOG.md:40]: bump of test dependency

File: CHANGELOG.md [hunk 8]
```diff
@@ -1,5 +1,49 @@
...
+* chore: bump github.com/gruntwork-io/terratest [#3222](https://github.com//kumahq/kuma/pull/3222)
```
Comment (jakubdyszkiewicz) [inline CHANGELOG.md:41]: this one too. And we had had terratest bump in the changelog

File: CHANGELOG.md [hunk 9]
```diff
@@ -1,5 +1,49 @@
...
+* build: fix dev_mac [#3209](https://github.com//kumahq/kuma/pull/3209)
```
Comment (jakubdyszkiewicz) [inline CHANGELOG.md:42]: CI changes

File: CHANGELOG.md [hunk 10]
```diff
@@ -1,5 +1,49 @@
...
+* chore: install latest kubectl version [#3207](https://github.com//kumahq/kuma/pull/3207)
```
Comment (jakubdyszkiewicz) [inline CHANGELOG.md:43]: CI changes

File: CHANGELOG.md [hunk 11]
```diff
@@ -1,5 +1,49 @@
...
+* tests: fix golden files after Helm chart update [#3205](https://github.com//kumahq/kuma/pull/3205)
```
Comment (jakubdyszkiewicz) [inline CHANGELOG.md:44]: CI fix

File: CHANGELOG.md [hunk 12]
```diff
@@ -1,5 +1,49 @@
...
+Changes:
+* feat: add kubernetes tags automatically [#3439](https://github.com//kumahq/kuma/pull/3439)
+* perf: update Mesh and ServiceInsights only when really needed [#3463](https://github.com//kumahq/kuma/pull/3463)
+* perf: eliminate uneccessary JSON marshalling [#3483](https://github.com//kumahq/kuma/pull/3483)
+* feat: sidecar injection webhook based on labels [#3417](https://github.com//kumahq/kuma/pull/3417)
+* chore: upgrade gui to new version [#3454](https://github.com//kumahq/kuma/pull/3454)
+* test: fix postgress tests permissions [#3443](https://github.com//kumahq/kuma/pull/3443)
+* feat: add affinity to CP and Ingress pods [#3036](https://github.com//kumahq/kuma/pull/3036)
+  👍contributed by @andrey-dubnik
+* chore: bump github.com/gruntwork-io/terratest [#3433](https://github.com//kumahq/kuma/pull/3433)
```
[the remark this answers is not in the document]
Comment (bartsmykla) [reply CHANGELOG.md]: addressed

File: CHANGELOG.md (hunk 1 above)
Comment (bartsmykla) [reply CHANGELOG.md]: addressed

File: CHANGELOG.md [hunk 13]
```diff
@@ -1,5 +1,49 @@
...
+* chore: bump github.com/golang-jwt/jwt/v4 from 4.1.0 to 4.2.0 [#3432](https://github.com//kumahq/kuma/pull/3432)
+* feat: consolidate tokens logic to support expiration, rotation, revocation and RSA256
+* ci: fix failing mac CI because of unsupported instance size [#3408](https://github.com//kumahq/kuma/pull/3408)
```
[the remark this answers is not in the document]
Comment (bartsmykla) [reply CHANGELOG.md]: addressed

File: CHANGELOG.md (hunk 2 above)
Comment (bartsmykla) [reply CHANGELOG.md]: addressed

File: CHANGELOG.md (hunk 3 above)
Comment (bartsmykla) [reply CHANGELOG.md]: addressed

File: CHANGELOG.md (hunk 7 above)
Comment (bartsmykla) [reply CHANGELOG.md]: addressed

File: CHANGELOG.md (hunk 8 above)
Comment (bartsmykla) [reply CHANGELOG.md]: addressed

File: CHANGELOG.md (hunk 4 above)
Comment (bartsmykla) [reply CHANGELOG.md]: addressed

File: CHANGELOG.md (hunk 5 above)
Comment (bartsmykla) [reply CHANGELOG.md]: addressed

File: CHANGELOG.md (hunk 6 above)
Comment (bartsmykla) [reply CHANGELOG.md]: addressed

File: CHANGELOG.md [hunk 14]
```diff
@@ -1,5 +1,36 @@
...
+* feat: add kubernetes tags automatically [#3439](https://github.com//kumahq/kuma/pull/3439)
+* perf: update Mesh and ServiceInsights only when really needed [#3463](https://github.com//kumahq/kuma/pull/3463)
+* perf: eliminate uneccessary JSON marshalling [#3483](https://github.com//kumahq/kuma/pull/3483)
+* feat: sidecar injection webhook based on labels [#3417](https://github.com//kumahq/kuma/pull/3417)
+* chore: upgrade gui to new version [#3454](https://github.com//kumahq/kuma/pull/3454)
+* test: fix postgress tests permissions [#3443](https://github.com//kumahq/kuma/pull/3443)
+* feat: add affinity to CP and Ingress pods [#3036](https://github.com//kumahq/kuma/pull/3036)
+  👍contributed by @andrey-dubnik
+* chore: bump github.com/golang-jwt/jwt/v4 from 4.1.0 to 4.2.0 [#3432](https://github.com//kumahq/kuma/pull/3432)
+* feat: consolidate tokens logic to support expiration, rotation, revocation and RSA256
```
Comment (jakubdyszkiewicz) [inline CHANGELOG.md:16]: nit: missing PR

File: CHANGELOG.md (hunk 9 above)
Comment (bartsmykla) [reply CHANGELOG.md]: addressed

File: CHANGELOG.md (hunk 10 above)
Comment (bartsmykla) [reply CHANGELOG.md]: addressed

File: CHANGELOG.md [hunk 15]
```diff
@@ -1,5 +1,36 @@
...
+* fix:: simplify cluster creation with endpoints [#3403](https://github.com//kumahq/kuma/pull/3403)
```
Comment (jakubdyszkiewicz) [inline CHANGELOG.md:17]: nit: `fix::`

File: CHANGELOG.md (hunk 11 above)
Comment (bartsmykla) [reply CHANGELOG.md]: addressed

**Review by jakubdyszkiewicz — APPROVED** [review]

File: CHANGELOG.md (hunk 14 above)
Comment (bartsmykla) [reply CHANGELOG.md]: addressed

File: CHANGELOG.md (hunk 15 above)
Comment (bartsmykla) [reply CHANGELOG.md]: addressed

**Pull request merged**